src/som/interpreter/actors/AbstractPromiseResolutionNode.java

```diff
@@ -51,15 +63,19 @@ protected void chainPromise(final SResolver resolver,
       final boolean isBreakpointOnPromiseResolution) {
     assert resolver.assertNotCompleted();
     SPromise promiseToBeResolved = resolver.getPromise();
-
+    boolean breakpointOnResolution = isBreakpointOnPromiseResolution;
     synchronized (promiseValue) {
       Resolution state = promiseValue.getResolutionStateUnsync();
       if (SPromise.isCompleted(state)) {
         resolvePromise(state, resolver, promiseValue.getValueUnsync(),
-           isBreakpointOnPromiseResolution);
+            breakpointOnResolution);
       } else {
         synchronized (promiseToBeResolved) { // TODO: is this really deadlock free?
-          promiseToBeResolved.setTriggerResolutionBreakpointOnUnresolvedChainedPromise(isBreakpointOnPromiseResolution);
+          // TODO: check this condition.
```

Comment:
@ctrlpz you are saying I should check this condition?
I don't know what you want me to do here. The condition looks ok, but I have no idea what you were thinking when you added the todo. What circumstances are you not sure about? What situation might arise where this is problematic?

This TODO note doesn't tell me how I can help you here.